dll/cpl/mmsys/sounds.c

```diff
@@ -1113,6 +1117,90 @@ InitImageList(UINT StartResource,
 }
 
 
+/**
+ * @brief
+ * Get the duration of a waveform audio file.
+ *
+ * @param[in]  pFileName
+ * The file name or full path of the file.
+ *
+ * @return
+ * The duration of the sound, in milliseconds.
+ * Returns 0 in case of failure.
+ **/
+static
+ULONG
+GetSoundDuration(LPCWSTR pFileName)
+{
+    DWORD dwSize, dwError;
+    LPWSTR pszCmd;
+    WCHAR Buffer[32];
+    PWCHAR endPtr;
+
+    dwSize = (wcslen(pFileName) + 38 + 1) * sizeof(WCHAR);
+    pszCmd = HeapAlloc(GetProcessHeap(), 0, dwSize);
+    if (!pszCmd)
+        return 0;
+
+    StringCbPrintfW(pszCmd, dwSize, L"open \"%s\" type waveaudio alias SoundFile", pFileName);
+
+    dwError = mciSendStringW(pszCmd, NULL, 0, NULL);
```

Comment:
A question: Did you choose to use those baroque mciSendString commands, in order not to have to open and close an MCI device? (an alternate way would be as we do in mplay32, and use mci_open and mciSendCommand function.)

Reply:
Why avoid `mciSendString` ? I think it's simpler than `mciSendCommand` in this case.

Reply:
The strings wastes computing resource. The commands is more computer-friendly.

Reply:
OK.